src/Chainweb/Miner/Coordinator.hs

```diff
@@ -121,18 +122,27 @@ newtype PrevTime = PrevTime BlockCreationTime
 
 data ChainChoice = Anything | TriedLast ChainId | Suggestion ChainId
 
+-- | A `Miner` defined in the nodes `miners` list. Should be subject to Primed
+-- Coordination.
+--
+newtype Whitelisted = Whitelisted Miner
+
+whitelisted :: Either Miner Whitelisted -> Miner
+whitelisted (Left m) = m
+whitelisted (Right (Whitelisted m)) = m
+
 -- | Construct a new `BlockHeader` to mine on.
 --
 newWork
     :: LogFunction
     -> CoordinationMode
     -> ChainChoice
-    -> Miner
+    -> Either Miner Whitelisted
     -> PactExecutionService
     -> TVar PrimedWork
     -> Cut
     -> IO (T3 PrevTime BlockHeader PayloadWithOutputs)
-newWork logFun mode choice miner@(Miner mid _) pact tpw c = do
+newWork logFun mode choice eminer pact tpw c = do
```

Comment:
Is `mode` used anywhere in this function (except for the recursive call)?

Reply:
No we can take that out now, good catch.